packages/openid4vc-holder/src/presentations/selection/PexCredentialSelection.ts

```diff
@@ -0,0 +1,334 @@
+) {
+  // Check if there's a 'from'. If not the structure is not as we expect it
+  if (!submissionRequirement.from) {
+    throw new AriesFrameworkError("Missing 'from' in submission requirement match")
+  }
+
+  const selectedSubmission: PresentationSubmissionRequirement = {
+    name: submissionRequirement.name,
+    purpose: submissionRequirement.purpose,
+    isRequirementSatisfied: false,
+    submission: [],
+
+    // TODO: if there's no count, min, max should we then assume the number to include is 1?
+    // TODO: if there's no count, min, but there is a max. Should we assume the min is 0 or 1?
+    needsCount: submissionRequirement.count ?? submissionRequirement.min ?? 1,
+  }
+
+  const satisfiedSubmissions: SubmissionEntry[] = []
+  const unsatisfiedSubmissions: SubmissionEntry[] = []
+
+  for (const inputDescriptor of presentationDefinition.input_descriptors) {
+    // We only want to get the submission if the input descriptor belongs to the group
+    if (!inputDescriptor.group?.includes(submissionRequirement.from)) continue
+
+    const submission = getSubmissionForInputDescriptor(inputDescriptor, selectResults)
+
+    if (submission.verifiableCredentials.length >= 1) {
+      satisfiedSubmissions.push(submission)
+    } else {
+      unsatisfiedSubmissions.push(submission)
+    }
+
+    if (satisfiedSubmissions.length === selectedSubmission.needsCount) {
+      break
+    }
+  }
+
+  return {
+    ...selectedSubmission,
+
+    // If there's enough satisfied submissions, the requirement is satisfied
+    isRequirementSatisfied: satisfiedSubmissions.length === selectedSubmission.needsCount,
+
+    // if the requirement is satisfied, we only need to return the satisfied submissions
+    // however if the requirement is not satisfied, we include all entries so the wallet could
+    // render which credentials are missing.
+    submission:
+      satisfiedSubmissions.length === selectedSubmission.needsCount
+        ? satisfiedSubmissions
+        : [...satisfiedSubmissions, ...unsatisfiedSubmissions],
+  }
+}
+
+function getSubmissionForInputDescriptor(
+  inputDescriptor: InputDescriptorV1 | InputDescriptorV2,
+  selectResults: W3cCredentialRecordSelectResults
+): SubmissionEntry {
+  // https://github.com/Sphereon-Opensource/PEX/issues/116
+  // FIXME: the match.name is only the id if the input_descriptor has no name
+  // Find first match
+  const matches = selectResults.matches?.filter(
+    (m) =>
+      m.name === inputDescriptor.id ||
+      // FIXME: this is not collision proof as the name doesn't have to be unique
+      m.name === inputDescriptor.name
+  )
+
+  let name = inputDescriptor.name
+  // If there's no name on the input descriptor, but the id does not contain
+  // any special characters or numbers (so only letters and spaces),
+  // we will use a sanitized version of the id as the name
+  if (!name && inputDescriptor.id.match(/^[a-zA-Z ]+$/)) {
+    name = sanitizeString(inputDescriptor.id)
```

Comment:
I think this is seomthing that should be done by the user and not the framework. Might miss some details as to why we do this though.

Reply:
I ll leave a comment, and look at it later